pkg/gardenlet/controller/seed/seed_control.go

```diff
@@ -265,26 +257,21 @@ func (r *reconciler) reconcile(ctx context.Context, gardenClient client.Client,
 
 		parentLogMessage := "Can't delete Seed, because the following objects are still referencing it:"
 		if len(associatedShoots) != 0 {
-			message := fmt.Sprintf("%s Shoots=%v", parentLogMessage, associatedShoots)
-			log.Info(message)
-			r.recorder.Event(seed, corev1.EventTypeNormal, v1beta1constants.EventResourceReferenced, message)
+			log.Info("Cannot delete SEed because the following Shoots are still referencing it", "shoots", associatedShoots)
+			r.recorder.Event(seed, corev1.EventTypeNormal, v1beta1constants.EventResourceReferenced, fmt.Sprintf("%s Shoots=%v", parentLogMessage, associatedShoots))
 		}
 		if len(associatedBackupBuckets) != 0 {
-			message := fmt.Sprintf("%s BackupBuckets=%v", parentLogMessage, associatedBackupBuckets)
-			log.Info(message)
-			r.recorder.Event(seed, corev1.EventTypeNormal, v1beta1constants.EventResourceReferenced, message)
+			log.Info("Cannot delete SEed because the following BackupBuckets are still referencing it", "backupBuckets", associatedBackupBuckets)
+			r.recorder.Event(seed, corev1.EventTypeNormal, v1beta1constants.EventResourceReferenced, fmt.Sprintf("%s BackupBuckets=%v", parentLogMessage, associatedBackupBuckets))
 		}
 
 		return errors.New("seed still has references")
 	}
 
-	log.Infof("[SEED RECONCILE]")
-
 	if !controllerutil.ContainsFinalizer(seed, gardencorev1beta1.GardenerName) {
+		log.Info("Adding finalizer")
 		if err := controllerutils.StrategicMergePatchAddFinalizers(ctx, gardenClient, seed, gardencorev1beta1.GardenerName); err != nil {
-			err = fmt.Errorf("could not add finalizer to Seed: %s", err.Error())
-			log.Error(err)
-			return err
+			return fmt.Errorf("could not add finalizer to Seed: %s", err.Error())
```

Comment:
Wouldn't it be better to wrap the error here, even though it wasn't before?
```suggestion
			return fmt.Errorf("could not add finalizer to Seed: %w", err)
```

Although, I see that in other places where finallizers are modified, simply the error is returned without any additional text, so we could also make it like that here.